src/main/scripts/validate.js

```diff
@@ -19,45 +19,62 @@ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
+const path = require('path')
 const fs = require('fs');
 const ajv = require('ajv');
+const { parseLanguageTag, parsedTagToCLDRLocale } = require('./language-utilities.js')
 
-const DATA_PATH = "src/main/data/facilities.json";
-const DATA_SCHEMA_PATH = "src/main/schemas/facilities.schema.json";
+const DATA_PATH = "src/main/data/";
+const DATA_SCHEMA_PATH = "src/main/schemas/%s.schema.json";
 
 /* load and validate the registry */
 
-let registry = JSON.parse(
-  fs.readFileSync(
-    DATA_PATH
-  )
-);
+var validator_factory = new ajv();
 
-if (!registry) {
-  throw "Cannot load registry";
-}
+for (const dataFile of fs.readdirSync(DATA_PATH).filter(f => /.json$/.test(f))) {
+  console.log(`Checking ${dataFile}`)
+  const name = path.basename(dataFile, ".json")
+  const registry = JSON.parse(fs.readFileSync(path.join(DATA_PATH, dataFile)))
+  const schemaFile = DATA_SCHEMA_PATH.replace("%s", name)
+  const validator = validator_factory.compile(JSON.parse(fs.readFileSync(schemaFile)))
 
-var validator_factory = new ajv();
+  if (!validator(registry))
+    throw `${name} registry fails validation`
 
-let validator = validator_factory.compile(
-  JSON.parse(
-    fs.readFileSync(
-      DATA_SCHEMA_PATH
-    )
-  )
-);
-
-if (! validator(registry)) {
-  console.log(validator.errors);
-  throw "Registry fails validation";
-};
-
-/* is the registry sorted */
-
-for(let i = 1; i < registry.length; i++) {
-  if (registry[i-1].code >= registry[i].code) {
-    throw "Registry key " + registry[i-1].code + " is " +
-      ((registry[i-1].code === registry[i].code) ? "duplicated" : "not sorted");
+  if (name ==='studios' || name === 'facilities') {
```

Comment:
@palemieux curious if you think these registry-specific checks should be moved to their own module, e.g. `<name>-validate.js`.

Reply:
You read my mind.

Reply:
Take a look at the latest.